snooty/diagnostics.py

```diff
@@ -768,3 +768,39 @@ def __init__(
             start,
             end,
         )
+
+
+class InvalidOpenApiResponse(Diagnostic):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            "OpenAPI Versioning data is malformed. Please contact DOP team.",
+            start,
+            end,
+        )
+
+
+class InvalidVersion(Diagnostic, MakeCorrectionMixin):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        api_version: str,
+        major_versions: List[str],
```

Comment:
nit: Usually better to use `Sequence` in function parameters rather than `List` unless you need mutation. Sequence can be any kind of... well, sequence type, and is immutable.

Reply:
Ah, just realized that did_you_mean is only accepting List[str] as input. How do we feel about leaving this? Or about changing did_you_mean to accept Sequences as well?

Reply:
Ah, forgot about `did_you_mean()`. That interface probably shouldn't be mutable, but it's fine; just return `list(self.major_versions)`